### PR TITLE
SONARIAC-462 Docker: parse WORKDIR instruction

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/WorkdirTree.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.api;
+
+import java.util.List;
+
+public interface WorkdirTree extends InstructionTree {
+  SyntaxToken instructionKeyword();
+  List<SyntaxToken> workdirList();
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/WorkdirTree.java:26]: IntelliJ underlines `workdirs` as Typo, so I named it `workdirList`. If you think `workdirs` is better then I can change it.

[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/WorkdirTree.java:26]: Fixed, but personally, I don't think it's better now, Java is so verbose in this case.

```
    List<SyntaxToken> children = tree.children().stream()
      .map(c -> (SyntaxToken) c)
      .collect(Collectors.toList());
    assertThat(children.get(0).value()).isEqualTo("WORKDIR");
```

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/WorkdirTreeImplTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  @Test
+  void shouldCheckParseTree() {
+    WorkdirTree tree = DockerTestUtils.parse("WORKDIR /foo bar /baz", DockerLexicalGrammar.WORKDIR);
+    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.WORKDIR);
+    assertThat(tree.instructionKeyword().value()).isEqualTo("WORKDIR");
+    assertThat(tree.workdirList().stream().map(w -> w.value())).containsExactly("/foo", "bar", "/baz");
+    assertThat(((SyntaxToken)tree.children().get(0)).value()).isEqualTo("WORKDIR");
+    assertThat(((SyntaxToken)tree.children().get(1)).value()).isEqualTo("/foo");
+    assertThat(((SyntaxToken)tree.children().get(2)).value()).isEqualTo("bar");
+    assertThat(((SyntaxToken)tree.children().get(3)).value()).isEqualTo("/baz");
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/WorkdirTreeImplTest.java]: Fixed, but personally, I don't think it's better now, Java is so verbose in this case.

```
    List<SyntaxToken> children = tree.children().stream()
      .map(c -> (SyntaxToken) c)
      .collect(Collectors.toList());
    assertThat(children.get(0).value()).isEqualTo("WORKDIR");
```

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/WorkdirTree.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/WorkdirTree.java:26]: II guess IntelliJ would like to read `workDirs`

**Pull request merged**